### ISSUE TITLE
panos_commit does not work with api_key

### ISSUE THREAD

**Issue by johnavp1989** [PaloAltoNetworks/ansible-pan#62]
Using an api_key with panos_commit to a firewall returns the following error:

```An exception occurred during task execution. To see the full traceback, use -vvv. The error was: pan.xapi.PanXapiError: api_key or api_username and api_password arguments required
fatal: [lab-firewall]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible__ffhBr/ansible_module_panos_commit.py\", line 148, in <module>\n    main()\n  File \"/tmp/ansible__ffhBr/ansible_module_panos_commit.py\", line 128, in main\n    device = base.PanDevice.create_from_device(ip_address, username, password)\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 2970, in create_from_device\n    system_info = device.refresh_system_info()\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3379, in refresh_system_info\n    system_info = self.show_system_info()\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3336, in show_system_info\n    root = self.xapi.op(cmd=\"show system info\", cmd_xml=True)\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3201, in xapi\n    self._xapi_private = self.generate_xapi()\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3243, in generate_xapi\n    kwargs = {'api_key': self.api_key,\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3195, in api_key\n    self._api_key = self._retrieve_api_key()\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3327, in _retrieve_api_key\n    timeout=self.timeout\n  File \"/usr/local/lib/python2.7/dist-packages/pandevice/base.py\", line 3002, in __init__\n    pan.xapi.PanXapi.__init__(self, *args, **kwargs)\n  File \"/usr/local/lib/python2.7/dist-packages/pan/xapi.py\", line 182, in __init__\n    'api_password arguments required')\npan.xapi.PanXapiError: api_key or api_username and api_password arguments required\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}
	to retry, use: --limit @/home/jpetrini/git/networking/ansible/commit.retry
```

Here's the playbook:

```- name: Commit changes to firewall                                
  hosts: all
  connection: local
  gather_facts: False

  vars:
    api_key: "LUFRPT0xTUxncTBqeXByZmM3aVJWMU55L0RySzF6R2s9YnI5VkkvNnliK1k0eGMzbDdpOUFaUHA5YzdDNklKS2QyWkNTcTdyUmxkbz0="

  roles:
    - role: PaloAltoNetworks.paloaltonetworks

  tasks:
  - name: commit changes
    panos_commit:
      ip_address: "{{ panos_ip }}"
      api_key: "{{ api_key }}"
```

It's not clear in the documentation whether this should work or not as api_key is not listed in the available options.

![image](https://user-images.githubusercontent.com/8194692/36542254-51cce2fe-17ae-11e8-95c2-a32437ff8866.png)

However the option is shown in one of the examples. The provided example is using panorama though not a firewall directly so it's possible that this option is not expected to work with a firewall.

![image](https://user-images.githubusercontent.com/8194692/36542322-800e0332-17ae-11e8-867d-d58f24cc768c.png)

If this is not a bug then it's a feature request. API keys are supported for just about every other module and it's a shame to write a playbook based around an API key but then be forced to use password auth in order to commit the changes.

**Comment by shinmog** [comment]
The documentation for this module needs to be updated as the code does indeed accept the API key.  However, it doesn't use the API key when invoking this function as it should.  So the fix should be to change this:

```
    device = base.PanDevice.create_from_device(ip_address, username, password)
```

...to this (and also update the module's documentation):

```
    device = base.PanDevice.create_from_device(ip_address, username, password, api_key)
```

**Comment by ivanbojer** [comment]
Fixed in V1.0.7
